webextension/scripts/popup.js

```diff
@@ -139,11 +139,47 @@ function search_tweet() {
   })
 }
 
+function makeValidURL() {
+  if (!isValidUrl(searchValue)) {
+    searchValue = 'https://' + searchValue
+  }
+}
+
+function useSearchBoxValue() {
```

Comment:
Would be better if you can make this work by passing `searchValue` in to the function, rather than referencing it as a global. I think you can also change the code to avoid the recursive call to `useSearchBoxValue()` by moving the call to `makeValidURL()` to the front of the function (which should also return the valid URL instead of modifying a global.)

Reply:
Sorry, I misinterpreted the line above and moved the call to `useSearchBoxValue()` to `makeValidURL()` and couldn't find a way to return the value. Sorry.

Now I have moved the call to `makeValidURL()` to the desired place.